src/Models/Model.php

```diff
@@ -23,7 +23,7 @@ abstract class Model implements ModelInterface, Arrayable, JsonSerializable {
 	/**
 	 * The model properties assigned to their types.
 	 *
-	 * @var array<string,string|array>
+	 * @var array<string,string|array{0:string,1:mixed}>
```

Comment:
I would consider adding native typehintig in these properties.

`protected static array $properties = [];`

Those must be a certain type, so why not enforce it ?

Reply:
Good catch! Resolved in 7b1e4eb